File: sketch-zoom/src/main/java/me/panpf/sketch/zoom/block/BlockSortException.java
package me.panpf.sketch.zoom.block;

import androidx.annotation.NonNull;

import java.util.List;

import me.panpf.sketch.SketchException;

public class BlockSortException extends SketchException {
    @NonNull
    private List<Block> blockList;
    private boolean useLegacyMergeSort;

    public BlockSortException(@NonNull IllegalArgumentException cause, @NonNull List<Block> blockList, boolean useLegacyMergeSort) {
        super(cause);
        this.blockList = blockList;
        this.useLegacyMergeSort = useLegacyMergeSort;
    }

    @NonNull
    @Override
    public synchronized IllegalArgumentException getCause() {
        return (IllegalArgumentException) super.getCause();
    }

    @NonNull
    public List<Block> getBlockList() {
        return blockList;
    }

    public boolean isUseLegacyMergeSort() {
        return useLegacyMergeSort;
    }
}

File: sketch-zoom/src/main/java/com/github/panpf/sketch/zoom/block/BlockSortException.kt
package com.github.panpf.sketch.zoom.block

class BlockSortException(
    cause: IllegalArgumentException,
    val blockList: List<Block>,
    val isUseLegacyMergeSort: Boolean
) : Exception(cause) {

    @get:Synchronized
    override val cause: IllegalArgumentException
        get() = super.cause as IllegalArgumentException
}